pages/articles.jsx

```diff
@@ -133,73 +53,162 @@ export default function ArticlesPage({ allContentTypes }) {
+			data.map(async item => {
+				return await await getRepoData(`/${item.path}`, keepEncoded, signal)
+			})
+		)
+	}
+
+
+	async function _fetchMarkdownFiles(allContentTypes, signal) {
+		// Returns array of all decoded config data for ALL related CT's
+		const _relatedCTConfigDataArray = allContentTypes.filter(function(i){
+			if (router.query.group) {
+				return router.query.group === i.group
+			} else {
+				return router.query.type === i.label
+			}
+		// eg: {label: "Posts", id: "posts", path: "content/posts", ...}
+		})
+
+		// Take _relatedCTConfigDataArray as a reference for the CT md files
+		// Check the returned md files and display only those which content_type matches to the id
+		// Get the CTs IDs to check for
+		const _idFromCTsConfigData = _relatedCTConfigDataArray.map(ct => ct.id)
+
+		// Returns a list of only the content directories ('path') from the config array
+		const _articlesLocationsList = new CT(_relatedCTConfigDataArray).getPaths()
+
+		// fetch all the files from the directories mentioned in `_articlesLocationsList`
+		listMarkdownFiles(_articlesLocationsList, false, signal)
+			.then(markdownFilesList => {
+				// flattern the output with the list of md files from the directory
+				const markdownFilesListFormatted = markdownFilesList.flat()
+
+				// Get the content from the md file
+				getMarkdownFileContent(markdownFilesListFormatted, false, signal)
+					.then(markdownFilesContent => {
+						// format md file content into the object
+						const formattedMarkdownFilesContent = markdownFilesContent.map(content =>
+							content && matter(content)
+						)
+
+
+						// Check if the `id` from the ContentType yaml file matches the `content_type` in the md file
+						const articleContents = []
+						_idFromCTsConfigData.forEach(id => {
+							formattedMarkdownFilesContent.forEach(markdownContent => {
+								if (id === markdownContent.data.content_type) {
+									articleContents.push(markdownContent)
+								}
+							})
+						})
+
+
+						// ADD `path` property to the markdown content `articleContents`
+						articleContents.forEach((item, index) => {
+							item.path = markdownFilesListFormatted[index].path
+						})
+
+						// ADD `name` property inside the `data` object
+						articleContents.forEach((item, index) => {
+							// Delete the extension
+							let fileName = markdownFilesListFormatted[index].name.split(`.`)
+							// Delete the last item in an array (extension)
+							fileName.pop()
+							// Concatenate all the rest items in an array
+							fileName = fileName.join()
+							// Create a `name` property at the `data`
+							item.data.name = fileName
+						})
+						setMarkdownContent(articleContents)
+					})
+					.catch(error => {
+						console.log(error)
+					})
+			})
+			.catch(error => {
+				console.log(error)
+			})
+	}
+
+
+	// Render errors for the user if the CT has missing REQUIRED fields
+	function _renderContentTypesErrors(contentTypesErrors) {
```

Comment:
@IvaTsu I think it would be better to make this a separate component and handle this from `_app.jsx` because if there are errors with the CT configs, it's going to affect usage of the whole app. Maybe errors specific to articles could be handled on this page.